### ISSUE TITLE
Preparing PR for upstream of new tasks api

### ISSUE THREAD

**Issue by sliverc** [adfinis-forks/aptly#23]
In this fork tasks API has been developed in a fully non lockable API. Therefore https://github.com/smira/aptly/pull/459 needs to be updated to reflect full new task api with running tasks in parallel.

This MR should not include mirror api or gpg api which need to be requests in other PRs.

**Comment by sliverc** [comment]
See https://github.com/smira/aptly/pull/459
